roles/contrail_node/tasks/main.yaml

```diff
@@ -0,0 +1,47 @@
+---
+- name: Untar Contrail tar-file
+  unarchive:
+    src: "{{ contrail_docker_images_path }}/contrail-kubernetes-docker-images_{{ contrail_version }}.tgz"
+    dest: "/tmp"
+  when: docker_registry is not defined
+
+- name: Load Contrail-vRouter-Agent image
+  docker_image:
+    name: contrail-agent
+    tag: "{{ contrail_version }}"
+    timeout: 300
+    load_path: "/tmp/contrail-agent-{{ os_release }}-{{ contrail_version }}.tar.gz"
+  when: docker_registry is not defined
+
+- name: Load Contrail-Kubernetes-Agent image
+  docker_image:
+    name: contrail-k8s-agent
+    tag: "{{ contrail_version }}"
+    timeout: 300
+    load_path: "/tmp/contrail-kubernetes-agent-{{ os_release }}-{{ contrail_version }}.tar.gz"
+  when: docker_registry is not defined
+
+- name: Add contrail service account to privileged scc
+  shell: oadm policy add-scc-to-user privileged system:serviceaccount:kube-system:contrail
+  delegate_to: "{{ groups.oo_first_master.0 }}"
+  when: inventory_hostname == groups.oo_first_master.0
```

Comment:
Can you put all tasks with the `inventory_hostname == contrail_first_master` under a block with a single condition?